modules/ichorcna/1.0/ichorcna.smk

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env snakemake
+
+
+# ---------------------------------------------------------------------------- #
+##### ATTRIBUTION #####
+# ---------------------------------------------------------------------------- #
+
+# Original snakemake author: Jasper Wong
+# Module author: Jasper Wong
+# Additional contributors: N/A
+
+
+# ---------------------------------------------------------------------------- #
+##### SETUP #####
+# ---------------------------------------------------------------------------- #
+
+### Modules ###
+
+import pandas as pd
+import numpy as np
+import oncopipe as op
+import glob
+import os
+
+### Directories ###
+
+resultsDir = "results/"
+logDir = "log/"
+
+
+# Setup module and store module-specific configuration in `CFG`.
+CFG = op.setup_module(
+    name = "ichorcna", 
+    version = "1.0",
+    subdirectories = ["inputs", "readDepth", "seg", "outputs"]
+)
+
+localrules:
+    _ichorcna_input_bam,
+    _ichorcna_output,
+    _ichorcna_all
+
+# ---------------------------------------------------------------------------- #
+##### RULES #####
+# ---------------------------------------------------------------------------- #
+
+CFG["runs"]["binSize"] = str(CFG["options"]["readcounter"]["binSize"])
```

Comment:
what does this do? I understand that it basically tells rule all the binsize for each sample. If it is the same for all samples, you can in the rule all just use `str(CFG["options"]["readcounter"]["binSize"])` and then you can multiply itty the number of samples, for example `*len(CFG["runs"]["tumour_sample_id"]` so it expands the length for each sample. There is example in the rule all of  module liftover

Reply:
Your interpretation is correct - I modified it to your method from liftOver.